### DISCUSSION TITLE
Write File Anywhere is the System

### DISCUSSION THREAD

**Discussion by markmccoid** [tauri-apps/tauri#9041] [Q&A] [answered]
The app I am trying to build is reading folders (could be located anywhere on the user's computer) and then building a json file describing each folder and saving it in that folder root.

It has to be in that directory because the json file is used by another application.

I am wondering if I'm missing some docs, because this seems like a common need.

I'm not sure if there is some config item that will allow the fs module to access or if this needs to be written in Rust.

Thanks for any guidance.

**Comment by AwaisCreativeDev** [comment]
In backend main.rs:
<pre><code>
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]



use std::fs::write;

#[tauri::command]
fn write_file(name: String, content: String) {
   let _ = write(name, content);
}

fn main() {
   tauri::Builder::default()
   .invoke_handler(tauri::generate_handler![write_file])
   .run(tauri::generate_context!())
   .expect("error while running tauri application");
}


</code></pre>

In frontend:
<pre><code>
import { invoke } from "@tauri-apps/api/tauri";

const writeFile = (name, content) => {
   invoke("write_file", { name, content });
}

let data = JSON.stringify({});

writeFile("./path/to/write/index.json", data);

</code></pre>

**Comment by chippers** [comment, marked as answer]
You can use the dialog plugin to achieve this, where the user gets to choose what directory to select.

https://v2.tauri.app/reference/javascript/dialog/#open

**Reply by AwaisCreativeDev** [reply]
thanks
